modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -1160,6 +1292,29 @@ public boolean doSegmentTest(Application.Name applicationName, Experiment.Label
         return queueLengthMap;
     }
 
+    /**
+     * Clear metadata cache if it is enabled.
+     */
+    @Override
+    public void clearMetadataCache() {
+        if(metadataCacheEnabled) {
```

Comment:
@bargenilesh Does this method clear the meta cache? When there is a batch assignment call in the middle of clearing the cache, does it refresh the cache with latest meta data? Can you explain how it works?